### PR TITLE
QEMU RockyLinux 9 and Ubuntu 23.04

### PR DESCRIPTION
**Pull request by feitnomore**
What this PR does / why we need it:
This PR adds RockyLinux 9 as well as Ubuntu 23.04 LTS support for QEMU.

Which issue(s) this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged): Fixes #

**Additional context**
This PR adds support for Rocky Linux 9.2 and Ubuntu 23.04 LTS for QEMU CAPI.
@AverageMarcus F.Y.I.

### CODE REVIEW

File: images/capi/packer/qemu/packer.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
         "{{user `boot_command_suffix`}}"
       ],
       "boot_wait": "{{user `boot_wait`}}",
+      "cpu_model": "host",
```
Comment (AverageMarcus) [inline images/capi/packer/qemu/packer.json:11]: Will this effect other OSs? This file is used by all the QEMU builds.

File: images/capi/packer/qemu/qemu-ubuntu-2204.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "boot_command_prefix": "c<wait>linux /casper/vmlinuz --- autoinstall ds='nocloud-net;s=http://{{ .HTTPIP }}:{{ .HTTPPort }}/22.04/'<enter><wait>initrd /casper/initrd<enter><wait>boot<enter>",
+  "boot_command_prefix": "c<wait>linux /casper/vmlinuz --- autoinstall ds='nocloud-net;s=http://{{ .HTTPIP }}:{{ .HTTPPort }}/22.04/'<enter><wait>initrd /casper/initrd<enter><wait><wait><wait>boot<enter>",
```
Comment (AverageMarcus) [inline images/capi/packer/qemu/qemu-ubuntu-2204.json:2]: Is this related to this PR?

File: images/capi/packer/qemu/qemu-ubuntu-2304.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+  "boot_command_prefix": "c<wait>linux /casper/vmlinuz --- autoinstall ds='nocloud-net;s=http://{{ .HTTPIP }}:{{ .HTTPPort }}/23.04/'<enter><wait><wait><wait><wait>initrd /casper/initrd<wait><wait><enter><wait>boot<enter>",
+  "build_name": "ubuntu-2304",
+  "distribution_version": "2304",
+  "distro_name": "ubuntu",
+  "guest_os_type": "ubuntu-64",
+  "iso_checksum": "c7cda48494a6d7d9665964388a3fc9c824b3bef0c9ea3818a1be982bc80d346b",
+  "iso_checksum_type": "sha256",
+  "iso_url": "https://releases.ubuntu.com/23.04/ubuntu-23.04-live-server-amd64.iso",
```
Comment (AverageMarcus) [inline images/capi/packer/qemu/qemu-ubuntu-2304.json:9]: Does this ISO change with new updates to 23.04? If so the checksum will eventually not match, right?

**Review by AverageMarcus — COMMENTED** [review]
Looks good. Just a few small comments / questions.

Aside from that, could you please squash your commits before we merge? 😄

File: images/capi/packer/qemu/packer.json (hunk 1 above)
Comment (feitnomore) [reply images/capi/packer/qemu/packer.json:11]: So, I don't think this will really cause any side effect. The newer kernels were suffering from a `kernel panic` while booting the installation, and doing some Google research this was the solution I've found.
I've built `qemu-ubuntu-2004`, `qemu-ubuntu-2204`, `qemu-ubuntu-2304`, `qemu-rockylinux-8` and `qemu-rockylinux-9` on a KVM nested virtualization, and everything worked fine.

File: images/capi/packer/qemu/qemu-ubuntu-2204.json (hunk 2 above)
Comment (feitnomore) [reply images/capi/packer/qemu/qemu-ubuntu-2204.json:2]: So, while testing, depending on the performance, I've got issues. If you are using a slower system, you may need to wait a bit more. In my case, I needed to add this, as well as increase some timeouts on `ansible.cfg` so that I could have success. The `ansible.cfg` changes I have not commited.

File: images/capi/packer/qemu/qemu-ubuntu-2304.json (hunk 3 above)
Comment (feitnomore) [reply images/capi/packer/qemu/qemu-ubuntu-2304.json:9]: Unfortunately, I don't see a specific file for 23.04.1 or .2 as we have for 22. The only file I see is this one.

Comment (AverageMarcus) [reply images/capi/packer/qemu/qemu-ubuntu-2304.json:9]: 😞  This is starting to become a recurring issue for us. 

@mboersma did we ever come up with any way of handling this when there isn't an `old.ubuntu.com` link available? 

I find it so weird that Ubuntu doesn't produce static images for their current releases.

Comment (feitnomore) [reply images/capi/packer/qemu/qemu-ubuntu-2304.json:9]: https://releases.ubuntu.com/23.04/SHA256SUMS might be a way. A possible solution could be to load the values from that file dynamically. 
```
curl -sS https://releases.ubuntu.com/23.04/SHA256SUMS | grep "ubuntu-23.04-live-server-amd64.iso" | awk '{print $1}'
```

Comment (AverageMarcus) [reply images/capi/packer/qemu/qemu-ubuntu-2304.json:9]: I'm more concerned about reproducible builds. If the ubuntu iso is updated with a breaking change it could break for image-builder users without there being any changes / releases to image-builder.

We currently doing have a way around this I don't think, it's just really annoying. 😞

**Comment by AverageMarcus** [comment]
I'm not going to block based on the iso issue as we have that elsewhere already. I'll open an issue to address the problem more generally.

/lgtm

@feitnomore before approval could you please squash the commits? :)
